firedrake/mesh.py

```diff
@@ -2105,6 +2106,10 @@ def VertexOnlyMesh(mesh, vertexcoords, missing_points_behaviour=None):
         that they have the same list of vertices (else the test is not
         possible): this operation scales with number of vertices and number of
         ranks.
+    :kwarg tolerance: the amount by which the local coordinates of a point are
+        allowed to fall outside the cell while still having the point count as
+        in the cell. Increase the default (1.0e-14) somewhat if vertices are
+        being lost in the :class:`VertexOnlyMesh` construction process.
```

Comment:
I think we need to note that this only works for cell boundaries within the (I presume on-process) mesh boundaries. I presume it can't guard against points on the mesh boundaries?

Reply:
If we could guard against being on the mesh boundary, we'd fix that case too!

Reply:
Okay then I think it just needs to be noted that it won't work if the lost point is on the mesh boundary

Reply:
I've added a couple of words to the comment to make it clear that this is for points inside the domain.